src/peft/tuners/boft/layer.py

```diff
@@ -510,9 +510,9 @@ def merge(self, safe_merge: bool = False, adapter_names: Optional[list[str]] = N
                     # because of the copy operation.
                     orig_weight = base_layer.weight.data.clone()
                     butterfly_oft_mat, boft_s = self.get_delta_weight(active_adapter)
-                    orig_weight = torch.transpose(orig_weight, 0, 1)
-                    orig_weight = torch.mm(butterfly_oft_mat, orig_weight)
-                    orig_weight = torch.transpose(orig_weight, 0, 1)
+                    orig_weight = torch.transpose(orig_weight, 0, 1).contiguous()
+                    orig_weight = torch.mm(butterfly_oft_mat, orig_weight).contiguous()
+                    orig_weight = torch.transpose(orig_weight, 0, 1).contiguous()
```

Comment:
I wonder if we can simplify this by only calling `.contiguous()` at the very last step, i.e. line 532:

```diff
- self.base_layer.weight.data = orig_weight
+ self.base_layer.weight.data = orig_weight.contiguous()
```

Same for line 532. And since there is also `Conv2d`, should the call be added there as well? So that would be lines 820 and 834.